### ISSUE TITLE
Settings.sample.php interferes with installer boardurl setting

### ISSUE THREAD

**Issue by Frenzie** [elkarte/Elkarte#2469]
Tested on the latest trunk from Git. This problem doesn't exist in 1.0.6.

Steps to reproduce:

1. Start the installer, for instance from http://localhost/elkarte-1.1
2. Fill out a different URL when prompted, for instance http://your-hostname.local/elkarte-1.1
3. Although the boardurl is set correctly, the theme URL will point to http://127.0.0.1/elkarte

**Comment by Frenzie** [comment]
NB I haven't debugged this or anything, but I imagine the issue arises because populating the database is the first step of the process.

**Comment by emanuele45** [comment]
That reminds me about the usual opcache problem resurfacing again...
Would you mind trying to change these lines in install.php:
```
	if (function_exists('opcache_invalidate'))
		opcache_invalidate(dirname(__FILE__) . '/Settings.php');
```
to:
```
	if (function_exists('opcache_invalidate'))
		opcache_invalidate(dirname(__FILE__) . '/Settings.php', true);
```

and test again the install?

**Comment by Frenzie** [comment]
tl;dr opcache is the problem, but that isn't the fix

Did they change the default? php.ini suggests the default is off and I didn't change it.

```
$ < /etc/php5/apache2/php.ini grep opcache
[opcache]
;opcache.enable=0
```

However, PHP info says it's up and running. I have to apologize, for I should've mentioned my PHP version is 5.6.19-2 (the latest on Debian/Stretch). Anyway, changing that line so that it always invalidates the cache in `CommonCode.php` doesn't solve the problem. Like I said, this only happens for me on 1.1, not on 1.0.6.

The solution then, if I understand your theory correctly, would probably be located around [this area](https://github.com/elkarte/Elkarte/blob/db7bf6c6c3214dc52c75fcddf22bf6c5450ed28c/install/Install_Controller.php#L684).

Incidentally, I now know how I came up with the nonsense above. It says `Step one. Populate database.` [in the comment](https://github.com/elkarte/Elkarte/blob/db7bf6c6c3214dc52c75fcddf22bf6c5450ed28c/install/Install_Controller.php#L714), but it means step one of the databasePopulation function, not step one of the entire process.

Anyway, running another test with opcache disabled in php.ini shows that it is indeed the root of the problem.

**Comment by emanuele45** [comment]
Opcache is enabled by default starting from 5.6 IIRC.
Interesting the difference between 1.0 and 1.1, in fact the two share basically the same code, just moved around a bit..

Could it be opcache is enabled, but opcache_invalidate function is not set?
Can you try it out just to rule it out?
```php
<?php

if (function_exists('opcache_invalidate'))
    echo 'exists';
else
    echo 'DARN!';
```

**Comment by Frenzie** [comment]
The function exists, both with opcache enabled and disabled.

**Comment by emanuele45** [comment]
AHA!
Try this:
```php
    if (function_exists('opcache_invalidate'))
        opcache_invalidate(TMP_BOARDDIR . '/Settings.php', true);
```

**Comment by Frenzie** [comment]
That's the ticket. Some leftover from changing stuff I imagine?

**Comment by emanuele45** [comment]
Yep, in 1.0 install.php is in the root, while in 1.1 is under ```/install/``` and ```dirname(__FILE__)``` points "obviously" to the install directory and not to the forum root, so the Settings.php cache is not invalidated.

**Comment by Frenzie** [comment]
Alright, who's committing the fix? =P

**Comment by emanuele45** [comment]
> Alright, who's committing the fix? =P
>

First comes, first serves? :P

**Comment by Frenzie** [comment]
lol, I'm not coming for another hour :P
